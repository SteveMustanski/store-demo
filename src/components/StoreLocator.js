import React, { Component } from 'react';
import Header from './Header';
import Button from './Button';
import Map from './Map';

class StoreLocator extends Component {
  render() {
    return (
      <div>
      <Header />
      <Button/> <Button/>
      <Map/>
      </div>
    )
  }
}

export default StoreLocator;