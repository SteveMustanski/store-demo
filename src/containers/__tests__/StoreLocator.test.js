import React from 'react';
import { shallow } from 'enzyme';
import StoreLocator from '../StoreLocator';
import axios from 'axios';


describe('StoreLocator', function () {
  let mountedStoreLocator;
  
  beforeEach(()=>{
    mountedStoreLocator = shallow(<StoreLocator />)
  });

  it('renders without crashing', () => {
   shallow(<StoreLocator />)
  });

  it('calls axios.get in #componentDidMount', ()=> {
    return mountedStoreLocator.instance().componentDidMount()
    .then(()=>{
      expect(axios.get).toHaveBeenCalled();
    });
  })

  it('calls axios.get with correct url', () => {
    return mountedStoreLocator.instance().componentDidMount()
    .then(()=> {
      expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/shops.json');
    })
  })

  it('renders a header', () => {
    const headers = mountedStoreLocator.find('Header');
    expect(headers.length).toBe(1);
  });

  it('renders 3 buttons', () => {
    const headers = mountedStoreLocator.find('Button');
    expect(headers.length).toBe(3);
  });

  it('renders a map', () => {
    const headers = mountedStoreLocator.find('Map');
    expect(headers.length).toBe(1);
  });

});

describe('choose map', ()=> {
  it('updates this.state.currentMap using the location that is passed to it', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
    let mockEvent = {target:{value:'testland'}};
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png');
  })
})