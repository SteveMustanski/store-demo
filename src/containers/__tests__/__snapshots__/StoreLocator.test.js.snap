// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders corretly 1`] = `[Function]`;
